### PR TITLE
[PkgConfigDeps] Creates `*.pc` files for `test_requires`

### LINKED ISSUE #11376: [question] PkgConfigDeps does not generate .pc files test_requires dependencies?
Using Conan version 1.48.1

I have a meson project which is dependent on a test dependency "PkgA"

After executing `conan install .` on my recipe

`.pc` file **not** generated when:

```
    generators = 'PkgConfigDeps'

    def build_requirements(self):
        self.test_requires('PkgA')
```

As a test, I changed it to a normal requires and the `.pc` file is generated:

```
    generators = 'PkgConfigDeps'
    requires = 'PkgA'
```

Is this expected or is there another way for me to get .pc files generated for test dependencies?

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Bugfix: `PkgConfigDeps` creates `*.pc` files for `test_requires`.
Docs: omit

Closes: https://github.com/conan-io/conan/issues/11376

### CODE REVIEW

File: conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py [hunk 1]
```diff
@@ -11,8 +11,13 @@ def __init__(self, conanfile):
     def content(self):
         """Get all the *.pc files content"""
         pc_files = {}
+        # Get all the dependencies
         host_req = self._conanfile.dependencies.host
-        for _, dep in host_req.items():
+        build_req = self._conanfile.dependencies.direct_build
+        test_req = self._conanfile.dependencies.test
+        for _, dep in list(host_req.items()) + list(build_req.items()) + list(test_req.items()):
```
Comment (SSE4) [inline conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py]: is there any risk of pkg-config files from different contexts overriding each other? like `build_requiries(protobuf)` + `requires(protobuf)`

Comment (franramirez688) [reply conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py]: Yes, but that's a problem that you could even have with other components. Have a look at this PR https://github.com/conan-io/conan/pull/10263 So, you should change that `pkg_config_name`/`pkg_config_aliases` property.

File: conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py [hunk 2]
```diff
@@ -621,5 +622,38 @@ def package_info(self):
...
+    PkgConfigDeps has to create any build and test requires declared on the recipe.
+
+    Related issue: https://github.com/conan-io/conan/issues/11376
+    """
+    client = TestClient()
+    app_folder = os.path.join(client.current_folder, "app")
+    test_folder = os.path.join(client.current_folder, "test")
+    os.makedirs(app_folder)
+    os.makedirs(test_folder)
+    with client.chdir(app_folder):
```
Comment (memsharded) [inline conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py]:
```suggestion
    with client.chdir("app"):
```

File: conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py [hunk 3]
```diff
@@ -621,5 +622,38 @@ def package_info(self):
...
+        client.run("new app/1.0 -m cmake_lib")
+        # client.run("new cmake_lib -d name=app -d version=1.0")
+        client.run("create .")
+    with client.chdir(test_folder):
```
Comment (memsharded) [inline conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py]:
```suggestion
    with client.chdir("test"):
```

File: conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py [hunk 4]
```diff
@@ -621,5 +622,38 @@ def package_info(self):
...
+        client.run("new test/1.0 -m cmake_lib")
+        # client.run("new cmake_lib -d name=test -d version=1.0")
+        client.run("create .")
+    # Create library having build and test requires
+    conanfile = textwrap.dedent(r'''
+        from conans import ConanFile
```
Comment (memsharded) [inline conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py]:
```suggestion
        from conan import ConanFile
```

File: conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py [hunk 5]
```diff
@@ -621,5 +622,38 @@ def package_info(self):
     client.save({"conanfile.txt": "[requires]\nhello/1.0\n"}, clean_first=True)
     client.run("install . --build=missing -g PkgConfigDeps")
     client.run_command("PKG_CONFIG_PATH=$(pwd) pkg-config --cflags hello")
-    assert r'flag2=\"my flag2\" flag1=\"my flag1\" -DUSER_CONFIG=\"user_config.h\" -DOTHER=\"other.h\"' in client.out
+    assert r'flag2=\"my flag2\" flag1=\"my flag1\" ' \
+           r'-DUSER_CONFIG=\"user_config.h\" -DOTHER=\"other.h\"' in client.out
 
+
+def test_pkgconfigdeps_with_build_and_test_requires():
```
Comment (memsharded) [inline conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py]: I would have preferred this is a faster integration test that checks the .pc is produced. Functional tests are slow and heavy, we want to keep our test suite as light as possible.

File: conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py (hunk 1 above)
Comment (memsharded) [inline conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py]: Not build_requires, only ``test_requires`` should generate .pc files

Build_requires are things like "cmake"

**Review by memsharded — CHANGES_REQUESTED** [review]

File: conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py (hunk 5 above)
Comment (franramirez688) [reply conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py]: I tend to say that a lot of tests from this module are integration ones...

File: conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py (hunk 1 above)
Comment (franramirez688) [reply conan/tools/gnu/pkgconfigdeps/pkgconfigdeps.py]: Ok. I saw CMakeDeps doing that so I only replicated that behavior

File: conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py (hunk 5 above)
Comment (memsharded) [reply conans/test/functional/toolchains/gnu/test_pkgconfigdeps.py]: It depends if they actually need the ``pkg_config`` tools or not. If not, yes, they can be moved to ``integration``. But not now, better in 2.0 only, and after we have stabilized and no need to do more develop->develop2 merges.

File: conans/test/integration/toolchains/gnu/test_pkgconfigdeps.py [hunk 6]
```diff
@@ -0,0 +1,629 @@
...
+    assert pc_files == ['OpenCL.pc', 'OtherCL.pc', 'pkgb.pc']
+    pc_content = client.load("OpenCL.pc")
+    assert "Name: OpenCL" in pc_content
+    assert "Description: Conan component: OpenCL" in pc_content
+    assert "Requires:" not in pc_content
+    pc_content = client.load("pkgb.pc")
+    assert "Requires: OpenCL" in get_requires_from_content(pc_content)
+
+
+def test_pkgconfigdeps_with_test_requires():
```
Comment (franramirez688) [inline conans/test/integration/toolchains/gnu/test_pkgconfigdeps.py:603]: @memsharded This is the new one. The rest of the diff is only a code movement.

**Review by memsharded — APPROVED** [review]

**Pull request merged**